pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
@smarterclayton I just tried to set the `requests` instead of `limits` in this PR and getting this when `oc deploy --latest`:

```
  39s		39s		2	{deployments-controller }			Warning		FailedRetry		history-limit-2: About to stop retrying history-limit-2: couldn't create deployer pod for test/history-limit-2: pods "history-limit-2-deploy" is forbidden: Failed quota: quota-terminating: must specify limits.cpu,limits.memory
```